### PR TITLE
Add smart-agent_nginx-ingress module

### PR DESCRIPTION
**Pull request by rsicart**
Create a dashboard for nginx ingress controller metrics.

### CODE REVIEW

File: modules/smart-agent_nginx-ingress/outputs.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+output "dashboard_group_id" {
+  description = "The ID of the nginx ingress dashboard group"
+  value       = signalfx_dashboard_group.nginx_ingress.id
+}
+
+output "dashboard_id" {
+  description = "The ID(s) of overview dashboard"
```
Comment (pdecat) [inline modules/smart-agent_nginx-ingress/outputs.tf:7]: There's only one:
```suggestion
  description = "The ID of the overview dashboard"
```

File: modules/smart-agent_nginx-ingress/main.tf [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+  time_range = 3600
+
+  on_chart_legend_dimension = "ingress"
+  color_by                  = "Dimension"
+  plot_type                 = "LineChart"
+
+  program_text = <<-EOF
+    data('nginx_ingress_controller_nginx_process_resident_memory_bytes').sum(by=['controller_class', 'controller_namespace']).publish()
+    EOF
+
```
Comment (pdecat) [inline modules/smart-agent_nginx-ingress/main.tf:119]:
```suggestion
```

File: modules/smart-agent_nginx-ingress/main.tf [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+  time_range = 3600
+
+  on_chart_legend_dimension = "ingress"
+  color_by                  = "Dimension"
+  plot_type                 = "LineChart"
+
+  program_text = <<-EOF
+    data('nginx_ingress_controller_nginx_process_cpu_seconds_total').sum(by=['controller_class', 'controller_namespace']).publish()
+    EOF
+
```
Comment (pdecat) [inline modules/smart-agent_nginx-ingress/main.tf:103]:
```suggestion
```

File: modules/smart-agent_nginx-ingress/main.tf [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+  time_range = 3600
+
+  on_chart_legend_dimension = "ingress"
+  color_by                  = "Dimension"
+  plot_type                 = "LineChart"
+
+  program_text = <<-EOF
+    data('nginx_ingress_controller_ingress_upstream_latency_seconds', rollup='delta').sum(by=['controller_class', 'controller_namespace', 'ingress']).publish()
+    EOF
+
```
Comment (pdecat) [inline modules/smart-agent_nginx-ingress/main.tf:87]:
```suggestion
```

File: modules/smart-agent_nginx-ingress/main.tf [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+  on_chart_legend_dimension = "ingress"
+  color_by                  = "Dimension"
+  plot_type                 = "LineChart"
+
+  program_text = <<-EOF
+    A = data('nginx_ingress_controller_requests', filter=filter('status', '5*') and filter('sfx_monitored', 'true'), rollup='delta', extrapolation='zero').sum(by=['controller_namespace', 'controller_class', 'ingress'])
+    B = data('nginx_ingress_controller_requests', filter('sfx_monitored', 'true'), rollup='delta', extrapolation='zero').sum(by=['controller_namespace', 'controller_class', 'ingress'])
+    ratio_50X = (A/B).scale(100).fill(value=0).publish('ratio_50X')
+    EOF
+
```
Comment (pdecat) [inline modules/smart-agent_nginx-ingress/main.tf:71]:
```suggestion
```

File: modules/smart-agent_nginx-ingress/main.tf [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+  on_chart_legend_dimension = "ingress"
+  color_by                  = "Dimension"
+  plot_type                 = "LineChart"
+
+  program_text = <<-EOF
+    A = data('nginx_ingress_controller_requests', filter=filter('status', '4*') and filter('sfx_monitored', 'true'), rollup='delta', extrapolation='zero').sum(by=['controller_namespace', 'controller_class', 'ingress'])
+    B = data('nginx_ingress_controller_requests', filter('sfx_monitored', 'true'), rollup='delta', extrapolation='zero').sum(by=['controller_namespace', 'controller_class', 'ingress'])
+    ratio_40X = (A/B).scale(100).fill(value=0).publish('ratio_40X')
+    EOF
+
```
Comment (pdecat) [inline modules/smart-agent_nginx-ingress/main.tf:53]:
```suggestion
```

File: modules/smart-agent_nginx-ingress/main.tf [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+  time_range = 3600
+
+  on_chart_legend_dimension = "ingress"
+  color_by                  = "Dimension"
+  plot_type                 = "LineChart"
+
+  program_text = <<-EOF
+    data('nginx_ingress_controller_requests', rollup='rate', extrapolation='zero').sum(by=['controller_class', 'controller_namespace', 'ingress']).publish()
+    EOF
+
```
Comment (pdecat) [inline modules/smart-agent_nginx-ingress/main.tf:35]:
```suggestion
```

File: modules/smart-agent_nginx-ingress/main.tf [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+  time_range = 3600
+
+  on_chart_legend_dimension = "ingress"
+  color_by                  = "Dimension"
+  plot_type                 = "LineChart"
+
+  program_text = <<-EOF
+    data('nginx_ingress_controller_requests').sum(by=['controller_class', 'controller_namespace', 'ingress']).publish()
+    EOF
+
```
Comment (pdecat) [inline modules/smart-agent_nginx-ingress/main.tf:19]:
```suggestion
```

File: modules/smart-agent_nginx-ingress/main.tf [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+  dashboard_group   = signalfx_dashboard_group.nginx_ingress.id
+  name              = "Nginx Ingress Controller"
+  time_range        = var.timerange
+
+  chart {
+    chart_id = signalfx_time_chart.nginx_ingress_current_requests.id
+    column   = 0
+    height   = 1
+    width    = 6
+    row      = 0
```
Comment (pdecat) [inline modules/smart-agent_nginx-ingress/main.tf:149]: I'd move row next to column for more readability:
```suggestion
    column   = 0
    row      = 0
    height   = 1
    width    = 6
```

**Review by pdecat — APPROVED** [review]

**Comment by rsicart** [comment]
Thanks for reviewing @pdecat !

**Comment by rsicart** [comment]
Hello all,

Do you have some feedback to share about this PR?

Thanks in advance

**Pull request merged**